src/features/todos.ts

```diff
@@ -1,8 +1,17 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction, Slice } from '@reduxjs/toolkit';
 import { Todo } from '../types/Todo';
+import { RootState } from '../app/store';
 
-export const todosSlice = createSlice({
+export const todosSlice: Slice<Todo[]> = createSlice({
   name: 'todos',
   initialState: [] as Todo[],
-  reducers: {},
+  reducers: {
+    setTodos(todos, action: PayloadAction<Todo[]>) {
+      todos.push(...action.payload);
```

Comment:
The `setTodos` reducer uses `push` to add todos, which may lead to incorrect state updates if the state is not empty. Consider using `todos = action.payload` to replace the current state with the new todos.